crates/astria-sequencer/src/authority/query.rs

```diff
@@ -0,0 +1,273 @@
+use astria_core::{
+    primitive::v1::ADDRESS_LEN,
+    protocol::abci::AbciErrorCode,
+};
+use cnidarium::Storage;
+use tendermint::abci::{
+    request,
+    response,
+    Code,
+};
+
+use crate::{
+    accounts::AddressBytes,
+    authority::state_ext::StateReadExt as _,
+};
+
+pub(crate) async fn validator_name_request(
+    storage: Storage,
+    request: request::Query,
+    params: Vec<(String, String)>,
+) -> response::Query {
+    let address = match preprocess_request(&params) {
+        Ok(address) => address,
+        Err(err) => return err,
+    };
+
+    let snapshot = storage.latest_snapshot();
+
+    let validator_set = match snapshot.get_validator_set().await {
+        Ok(validator_set) => validator_set,
+        Err(err) => {
+            return error_query_response(
+                Some(err),
+                AbciErrorCode::INTERNAL_ERROR,
+                "failed to get validator set",
+            );
+        }
+    };
+
+    if validator_set.get(address.address_bytes()).is_none() {
+        return error_query_response(
+            None,
+            AbciErrorCode::VALUE_NOT_FOUND,
+            "validator address not found in validator set",
+        );
+    }
```

Comment:
My thought for this was that it may be nice from a UX perspective to get an error saying the validator address wasn't in the set. If you simply made a query where you messed up a character of address, I figured that may be a nice thing to know, as opposed to just being told that the address doesn't have a name. Happy to remove though if you disagree

Reply:
Rearranged such that it first checks for the name, then checks the validator if the name is not found in 29496c7.